packages/http/src/clients/http.client.ts

```diff
@@ -135,6 +138,10 @@ export class HttpClient implements HttpClientInterface {
         if (this.isResponseError(updatedResponse) &&
             requestOptions.isRetryable && requestOptions.isRetryable(request, updatedResponse)
         ) {
+            for (let httpErrorResponseInterceptor of this.httpErrorResponseInterceptors) {
```

Comment:
You need to call the error interceptors whenever there's an error, not only when it's being retried.